### PR TITLE
adding HashiCorp Vault namespaces examples to KMS doc

### PR DESCRIPTION
**Pull request by nicolaka**
Signed-off-by: nicolaka <nicolaka@gmail.com>

Summary
I updated the KMS.md doc with additional details on using Vault Namespaces with cosign.

Ticket Link
Fixes

Release Note
NONE

### CODE REVIEW

**Comment by nicolaka** [comment]
I opened this new PR. I deleted old branches. Fetched current HEAD, and still seeing the same edits to `.github/workflows/scorecard_action.yml`.

```
 git log -3                                                                      
error: cannot run less: No such file or directory
commit 12b7b9cf957f16ae2a43646e755edc4366c96bff (HEAD -> update-vault-kms-doc, origin/update-vault-kms-doc)
Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Date:   Wed May 25 20:31:40 2022 +0200

    adding HashiCorp Vault namespaces examples to KMS doc
    
    Signed-off-by: nicolaka <nicolaka@gmail.com>
```

**Review by dlorenc — APPROVED** [review]

**Comment by hectorj2f** [comment]
@nicolaka there are some extra whitespaces that need to be removed.

File: .github/workflows/scorecard_action.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
           persist-credentials: false
 
       - name: "Run analysis"
-        uses: ossf/scorecard-action@c1aec4ac820532bab364f02a81873c555a0ba3a1
+        uses: ossf/scorecard-action@5c8bc69dc88b65c66584e07611df79d3579b0377
```
Comment (developer-guy) [inline .github/workflows/scorecard_action.yml:31]: I think this change is out of the scope of this PR, would you mind sending it by opening another PR, does it make sense?

File: KMS.md [hunk 2]
```diff
@@ -152,6 +152,38 @@ The URI format for Hashicorp Vault KMS is:
 This provider requires that the standard Vault environment variables (VAULT_ADDR, VAULT_TOKEN) are set correctly.
 This provider also requires that the `transit` secret engine is enabled
 
+If you enabled `transit` secret engine at different path with the use of `-path` flag (i.e., `$ vault secrets enable -path="someotherpath" transit`), you can use `TRANSIT_SECRET_ENGINE_PATH` environment variable to specify this path while generating a key pair like the following:
```
Comment (developer-guy) [inline KMS.md:155]: IIRC, @dlorenc would want to move documentation parts from cosign repository to the sigstore website which you can find [here](https://github.com/sigstore/sigstore-website), so, we added all the details related to Vault here, please do check it out first, then suggest your updates here.

👉 https://docs.sigstore.dev/cosign/kms_support#hashicorp-vault

**Review by developer-guy — CHANGES_REQUESTED** [review]

File: .github/workflows/scorecard_action.yml (hunk 1 above)
Comment (nicolaka) [reply .github/workflows/scorecard_action.yml:31]: this had been automatically been added to my commit. my local branch has same value of upstream. Are github actions mutating this ?

**Pull request closed without merging**